R/GatingSet2flowJo.R

```diff
@@ -270,17 +304,39 @@ transformationNode <- function(gh, matInfo){
                   )
 
                 }else if(trans.type == "logicle"){
-                  param <- as.list(environment(func))
-                  withBasis <- - 10 ^ (2 * as.vector(param[["w"]]))
-                  transNode <- xmlNode("biex"
-                                       , namespace = "transforms"
-                                       , attrs = c("transforms:length" = 4096
-                                                   , "transforms:maxRange" = param[["t"]]
-                                                   , "transforms:neg" = param[["a"]]
-                                                   , "transforms:width" = withBasis
-                                                   , "transforms:pos" = param[["m"]]
-                                       )
-                  )
+
+                  if(!is.null(Scaling)){
+                  # Added scaling adjustment
+                    offset <- 316
```

Comment:
The hard-coded value should be avoided

Reply:
First of all, I just want to say that I never meant for this change to be perfect for every case. It works for my case and the idea could be extended to other transformations. This is why I originally added this as an issue with explanations.

I needed something as a default if the next part of the code didn't work, or didn't have an input. 316 and 3162277 were the defaults for my project. I don't remember right now how to calculate the defaults. Again, once I got this working for my project I considered it done. The sample code of how I defined Scaling and then used it is below. The first three channels are linear and go through trans.type == "flowJo_linear", while the next 9 go through trans.type == "logicle".

        Scaling <- list(c(0,fsch.max), c(-100000,ssch.max), c(0,fsca.max), # "FSC-H" "SSC-H" "FSC-A"
                        c(316, 5000000), c(100,30000000), c(10, 5000000), # "BL1-H" "BL2-H" "BL5-H"
                        c(800,30000000), c(800,30000000), c(1,  5000000), # "RL1-H" "RL2-H" "VL1-H"
                        c(316,30000000), c(100,30000000), c(316,3162277) # "VL2-H" "VL4-H" "VL6-H"
        )

        names(Scaling) <- (gs@data[[1]]@parameters@data$name)

        GatingSet2flowJo(gs, output.path, Scaling)